app/validators/placeholders/placeholder_validator.py

```diff
@@ -60,6 +67,54 @@ def validate_placeholder_object(self, placeholder_object):
                 differences=placeholder_differences,
             )
 
+    def validate_placeholder_answer_id(self, argument):
+        """
+        Validate answer_id exists in Answer Context
+        Args:
+         argument: answer id
+        Returns:
+            True  : if exists
+            False : if not exists
+        """
+
+        if isinstance(argument, str):
+            answers = self.questionnaire_schema.answers_with_context
+            if argument not in answers:
+                self.add_error(
+                    ValueSourceValidator.ANSWER_REFERENCE_INVALID, identifier=argument
+                )
+                return False
+            return True
+
+    def validate_option_label_value_placeholder(self, argument):
+        """
+        validate answer_id  for transform 'option_label_from_value' is of type of ['Checkbox','Radio','Dropdown']
+        Args:
+         argument: answer id argument passed to transform
+
+        """
+        answers = self.questionnaire_schema.answers_with_context
+        answer_id_exists = self.validate_placeholder_answer_id(argument)
+
+        # if answer id doesn't exist , no further validation is done
+        if not answer_id_exists:
+            return
+
+        # if no "options" present , then it is not Answer Option type
+        if "options" in answers[argument]["answer"]:
+            if not any(
+                x.value == answers[argument]["answer"]["type"] for x in AnswerOptionType
+            ):
+                self.add_error(
+                    error_messages.ANSWER_OPTION_LABEL_VALUE_TYPE_INVALID,
+                    identifier=argument,
+                )
+        else:
+            self.add_error(
+                error_messages.ANSWER_OPTION_LABEL_VALUE_TYPE_INVALID,
+                identifier=argument,
+            )
+
```

Comment:
We just need to check if answer type is Checkbox/Radio/Dropdown. We don't need to check if options exists, cause if it is one of those 3, it will have options, if it didn't it won't get this far.

```py
        if not any(
            x.value == answers[argument]["answer"]["type"] for x in AnswerOptionType
        ):
            self.add_error(
                error_messages.ANSWER_OPTION_LABEL_VALUE_TYPE_INVALID,
                identifier=argument,
            )
```

---

Not sure if we even need `AnswerOptionType` enum class as this is specific to this and there is another answer type (`Relationship`) which has `options` but we don't support here. Could be just a set. Not a blocker though.

Reply:
i think we need `option` check because if we have valid `answer_id` which doesn't have `options` it will raise key error for `type` ?

Reply:
`type` is a key that exists in every answer.